### ISSUE TITLE
Meta Data not attached 

### ISSUE THREAD

**Issue by jeykodev** [dachcom-digital/pimcore-seo#10]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no

At first: thanks for your amazing bundles, it realy simplyfies our daily work! 😊 

**My problem:**

Just like in the offical [pimcore demo](https://demo.pimcore.fun/admin) I've build the following site structure in pimcores document tree: 

![image](https://user-images.githubusercontent.com/10532676/87755762-e6c1ac80-c807-11ea-9e19-eef8bb1914b0.png)

The root document (ID 1) ist our start page and localized pages are created under its dedicated language path. Like in the demo we created a link from the root of our main language `de` to the root document `/`. 

After setting up the bundle, everything works as expected but on the document root no meta data will be attached by the bundle. The reason for that is the `AutoMetaDataAttachListener.php`. The request is validated here so that the meta data will only be attached on certain conditions. 

The attachment fails here:
https://github.com/dachcom-digital/pimcore-seo/blob/01f773f6843f4fa5620dbb152f5384a7d4a64ddf/src/SeoBundle/EventListener/AutoMetaDataAttachListener.php#L87-L89

I think this is because the redirect from the language root to the document root. 

Can you please confirm that this is the expected behaviour? How can we set the meta data in this cases? Thank you.

**Comment by solverat** [comment]
@Homer0815: I think I need more background here. Which document exactly does not have any attached data in your setup?

**Comment by solverat** [comment]
@Homer0815 Can't reproduce it. I just added the same setup with some SEO data on "Start" (Root) which just works fine. Of course, there will be a redirect first from `/de` to `/` and at this time, there is no available document.

**Comment by jeykodev** [comment]
@solverat Thanks for your answer. I still can reproduce the issue and digged a little bit deeper into the problem

> @Homer0815: Which document exactly does not have any attached data in your setup?

The Problem is that no meta data will be attached on the document with the DocumentID=1, every other document works and there is no problem with any redirects (sorry that was a little missleading)

The cause for that can be found here: 

https://github.com/dachcom-digital/pimcore-seo/blob/01f773f6843f4fa5620dbb152f5384a7d4a64ddf/src/SeoBundle/EventListener/AutoMetaDataAttachListener.php#L103-L106

I don't know why, but the root document cant be resolved using the `DocumentResolver` and the `$document` will be `null`here. Maybe there is some correlation with some other modules in the project. I will try to reproduce it with a clean installation.

**Comment by jeykodev** [comment]
@solverat Sorry, I found the issue. I've had this action in my `DefaultController.php`.:

>/**
 >* Handling the welcome Action
 >* @ Route("/")
 >* @ Template("Pages/main-content.html.twig")
 >* @ param Request $request
 >*/
>public function defaultAction(Request $request)
>{
>
>}




Because of the `@Route` annotion the request was resolved in a different way than calling a document from the document tree. Therefore no document was attached to the request, so your `AutoMetaDataAttachListener.php` did not attach the meta data, because it had no document for reference.

Thank you anyway and keep up your great work. 😊
